src/TikTok/Provider.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace SocialiteProviders\TikTok;
+
+use Illuminate\Support\Arr;
+use Laravel\Socialite\Two\InvalidStateException;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'TIKTOK';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        'user.info.basic',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return 'https://open-api.tiktok.com/platform/oauth/connect?'.http_build_query([
+            'client_key'    => $this->clientId,
+            'state'         => $state,
+            'response_type' => 'code',
+            'scope'         => $this->formatScopes($this->getScopes(), $this->scopeSeparator),
+            'redirect_uri'  => $this->redirectUrl,
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function user()
+    {
+        if ($this->user) {
+            return $this->user;
+        }
+
+        if ($this->hasInvalidState()) {
+            throw new InvalidStateException();
+        }
+
+        $response = $this->getAccessTokenResponse($this->getCode());
+
+        $token = Arr::get($response, 'data.access_token');
+
+        $this->user = $this->mapUserToObject(
+            $this->getUserByToken([$token, Arr::get($response, 'data.open_id')])
```

Comment:
Can we key these array params? Just for the sake of readability as `'open_id'      => $data[1],` would be easier to interpret as `'open_id'      => $data['open-id'],`

 :) Thanks